File: euler/euler2.rb

def fiboeuler(limit)
  x,y = 1,2
  numary = [1]
  i = 1
  sum = 0

  while x < limit
    x,y = y, x+y
    numary << x

    if numary[i-1] % 2 == 0
      sum += numary[i-1]
    end
    i += 1
  end
  sum
end

puts fiboeuler(4000000)

File: euler/euler2.hs
{-
 - Return the sum of the even fibonacci numbers
 - that do not exceed four million.
 -}

fib :: [Int]
fib = 0 : scanl (+) 1 fib

fibsUpTo :: Int -> [Int]
fibsUpTo n = takeWhile (< n) fib

limit :: Int
limit = 4000000

main :: IO ()
main = do
    print $ sum $ filter even (fibsUpTo limit)

